src/components/views/dialogs/InviteDialog.tsx

```diff
@@ -199,17 +244,42 @@ interface IDMRoomTileProps {
     onToggle: (RoomMember) => any;
     highlightWord: string;
     isSelected: boolean;
+    isFavorite: boolean;   // determines whether or not member is your favorite
+    isAvailable: boolean;  // determines whether or not member role has a person fulfilling that role
+    roleCategoryId: string;
+    kind: string;
+    error: any;
 }
 
 class DMRoomTile extends React.PureComponent<IDMRoomTileProps> {
+    showDetailView = false;
+
     _onClick = (e) => {
         // Stop the browser from highlighting text
         e.preventDefault();
         e.stopPropagation();
 
+     if(!this.showDetailView)
         this.props.onToggle(this.props.member);
     };
 
+    onClickView(ev: React.MouseEvent<HTMLElement>) {
+        ev.preventDefault();
+        ev.stopPropagation();
+        this.showDetailView = !this.showDetailView;
+
+        const detailView = ev.currentTarget.parentNode.querySelector<HTMLElement>('#mx_table_role_detail');  // selects current detailed view
+        let viewDetailBtn = ev.currentTarget.parentNode.querySelector<HTMLElement>("#mx_viewDetailBtn");  // selects current button clicked
+        if (detailView.style.display === "none") {
+            detailView.style.display = "block";
+            viewDetailBtn.innerHTML = "Hide Detail";
```

Comment:
SHould "Hide detail" be just "Hide" to be consistent with the "View detail" changing to "View"?

Reply:
I dont think it will be clear to user so I left with Hide detail there we will ask audience later. View will make sense to view item in detail but then hide detail will make user completely understand what s/he is hiding. "Hide" term may imply user is hiding entire search item so I did not make that just "hide"

Reply:
View detail would sound larger text and also matrix uses just "View" for displaying room in another page from modal like for instance for "room directory"...so when view is expanded I wanted to make it clear to user you are hiding detail view by showing "hide detail". This will remove ambiguity.